packages/Kokkos/test/tstKokkosView.cpp

```diff
@@ -0,0 +1,266 @@
+    using DeviceType = typename Node::device_type;
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    // Create a view in the execution space.
+    using ViewType = Kokkos::View<Scalar *, DeviceType>;
+    const int size = 1000;
+    ViewType data( "data", size );
+
+    // Populate the view in the execution space.
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, size ),
+                          FillFunctor<ViewType>( data ) );
+    Kokkos::fence();
+
+    // Mirror the view to the host space and check the results.
+    typename ViewType::HostMirror host_data =
+        Kokkos::create_mirror_view( data );
+    for ( int i = 0; i < size; ++i )
+    {
+        TEST_EQUALITY( host_data( i ), i );
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Test assigning views with different layouts.
+TEUCHOS_UNIT_TEST_TEMPLATE_2_DECL( View, layout_assign_kernel, Scalar, Node )
+{
+    // Get types.
+    using DeviceType = typename Node::device_type;
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    // Create a view in the execution space.
+    using ViewType1 = Kokkos::View<Scalar **, Kokkos::LayoutLeft, DeviceType>;
+    const int size = 1000;
+    const int dim = 3;
+    ViewType1 data_1( "data_1", size, dim );
+
+    // Populate the first view.
+    for ( int d = 0; d < dim; ++d )
+    {
+        auto sv = Kokkos::subview( data_1, Kokkos::ALL(), d );
+        Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, size ),
+                              FillFunctor<decltype( sv )>( sv ) );
+        Kokkos::fence();
+    }
+
+    // Create another view.
+    using ViewType2 = Kokkos::View<Scalar **, Kokkos::LayoutRight, DeviceType>;
+    ViewType2 data_2( "data_2", size, dim );
+
+    // Copy the first view into the second.
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<ExecutionSpace>( 0, size ),
+        AssignFunctor<ViewType1, ViewType2>( data_1, data_2 ) );
+    Kokkos::fence();
+
+    // Check the second view on the host.
+    typename ViewType2::HostMirror host_data =
+        Kokkos::create_mirror_view( data_2 );
+    for ( int i = 0; i < size; ++i )
+    {
+        for ( int d = 0; d < dim; ++d )
+        {
+            TEST_EQUALITY( host_data( i, d ), i );
+        }
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Test creating a view and run a basic reduction kernel.
+TEUCHOS_UNIT_TEST_TEMPLATE_2_DECL( View, basic_reduce_kernel, Scalar, Node )
+{
+    // Get types.
+    using DeviceType = typename Node::device_type;
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+    // Create a view in the execution space.
+    using ViewType = Kokkos::View<Scalar *, DeviceType>;
+    const int size = 1000;
+    ViewType data( "data", size );
+
+    // Populate the view in the execution space.
+    Kokkos::parallel_for( Kokkos::RangePolicy<ExecutionSpace>( 0, size ),
```

Comment:
I put it after because it said this on the kokkos page:

```
To avoid that you either need to 
call Kokkos::Cuda::fence() (or just Kokkos::fence()), after kernels, or
you can set the environment variable CUDA_LAUNCH_BLOCKING=1.
```